tests/test_prepare_recalibration.yml

```diff
@@ -45,17 +68,31 @@
     - preprocessing
   files:
     - path: results/csv/variantcalled.csv
+      md5sum: 4d0effd3d8dc2b814230a189e7ca9dba
     - path: results/multiqc
     - path: results/preprocessing/markduplicates/test/test.md.cram
+    # binary changes md5sums on reruns.
     - path: results/preprocessing/markduplicates/test/test.md.cram.crai
+    # binary changes md5sums on reruns.
     - path: results/reports/bcftools/strelka/test/test.strelka.variants.bcftools_stats.txt
+      md5sum: 0741b5dd5e6c5428bd5d4f1b7a923dd5
     - path: results/reports/vcftools/strelka/test/test.strelka.variants.FILTER.summary
+      md5sum: 39ff2cc8eb7495a14a6b76e0ab627027
     - path: results/reports/vcftools/strelka/test/test.strelka.variants.TsTv.count
+      md5sum: ee7dafc8d941b8502a04a63dc3126fff
     - path: results/reports/vcftools/strelka/test/test.strelka.variants.TsTv.qual
+      # The text-based file test.strelka.variants.TsTv.qual has different md5sums on my linux system and the GitHub-test-server.
```

Comment:
That is very striking! Shouldn't the use of a container lead to numerically stable results? 🤔

Reply:
Yeah, it is strange. This content of the file  `results/reports/vcftools/strelka/test/test.strelka.variants.TsTv.qual` when running the command `nextflow run main.nf -profile test,prepare_recalibration_bam,skip_bqsr,singularity --tools strelka` on my own server:
```
QUAL_THRESHOLD  N_Ts_LT_QUAL_THRESHOLD  N_Tv_LT_QUAL_THRESHOLD  Ts/Tv_LT_QUAL_THRESHOLD N_Ts_GT_QUAL_THRESHOLD  N_Tv_GT_QUAL_THRESHOLD  Ts/Tv_GT_QUAL_THRESHOLD
0       0       0       -nan    3       4       0.75
2       0       1       0       2       4       0.5
5       1       1       1       2       3       0.666667
21      1       2       0.5     2       2       1
28      1       3       0.333333        2       1       2
33      1       4       0.25    2       1.3526e-319     inf
42      1       5       0.2     1       1.3526e-319     inf
391     2       5       0.4     4.00193e-322    1.3526e-319     0.00295869
```
I'll try to see if I can identify more precisely what the problem is.

Reply:
The test for all the lines in `test.strelka.variants.TsTv.qual`, that is,
```
      contains:
        [
          "QUAL_THRESHOLD	N_Ts_LT_QUAL_THRESHOLD	N_Tv_LT_QUAL_THRESHOLD	Ts/Tv_LT_QUAL_THRESHOLD	N_Ts_GT_QUAL_THRESHOLD	N_Tv_GT_QUAL_THRESHOLD	Ts/Tv_GT_QUAL_THRESHOLD",
          "2	0	1	0	2	4	0.5",
          "5	1	1	1	2	3	0.666667",
          "21	1	2	0.5	2	2	1",
          "28	1	3	0.333333	2	1	2",
          "33	1	4	0.25	2	1.3526e-319	inf",
          "42	1	5	0.2	1	1.3526e-319	inf",
          "391	2	5	0.4	4.00193e-322	1.3526e-319	0.00295869",
        ]
```
fails for the three lines containing numbers with "scientific" notation:
```
=================================== FAILURES ===================================
_________________________________ test session _________________________________
'33	1	4	0.25	2	1.3526e-319	inf' was not found in /tmp/pytest_workflow_jqwdxbx5/Run_prepare_recalibration_starting_from_bam_and_skip_baserecalibration/results/reports/vcftools/strelka/test/test.strelka.variants.TsTv.qual while it should be there.
_________________________________ test session _________________________________
'42	1	5	0.2	1	1.3526e-319	inf' was not found in /tmp/pytest_workflow_jqwdxbx5/Run_prepare_recalibration_starting_from_bam_and_skip_baserecalibration/results/reports/vcftools/strelka/test/test.strelka.variants.TsTv.qual while it should be there.
_________________________________ test session _________________________________
'391	2	5	0.4	4.00193e-322	1.3526e-319	0.00295869' was not found in /tmp/pytest_workflow_jqwdxbx5/Run_prepare_recalibration_starting_from_bam_and_skip_baserecalibration/results/reports/vcftools/strelka/test/test.strelka.variants.TsTv.qual while it should be there.```

Reply:
The md5sums of `results/reports/vcftools/strelka/test/test.strelka.variants.TsTv.qual` does seem to be stabile. The file is just generated for two different tests, and that results in different md5sums. I guess I just didn't pay enough attention to that, and perhaps got the md5sums mixed up 🤦‍♂️